lib/Ix/DBIC/ResultSet.pm

```diff
@@ -515,6 +516,15 @@ sub _ix_wash_rows ($self, $rows) {
     for my $key ($by_type{boolean}->@*) {
       $row->{$key} = $row->{$key} ? $true : $false if defined $row->{$key};
     }
+
+    for my $key ($by_type{datetime}->@*) {
+      if ($row->{$key}) {
+        # Doesn't already look like an RFC3339 Zulu date?
+        if ($row->{$key} !~ /Z/) {
+          $row->{$key} = parsepgdate($row->{$key});
+        }
+      }
+    }
```

Comment:
So, this code (wash_rows) runs on data that we're going to turn return in a method result payload.  It does stuff like, as above, turn `1` in a boolean column into a `true`.  I think what we want is to make sure that any datetime values are Ix::DateTime objects.  Then, perl consumers of results are guaranteed to have objects and JSON will get the formatted data.

Does this check really account for the possible things in a row?  That is: this code seems predicated on the idea that any value in `$row->{datetime_field}` is either undef, a 3339 string, or a pgsql string.  Are those the right options?

I'm not sure, I just want to make sure you thought it through.  Something about this code makes me go "hm."